kani-compiler/src/kani_middle/transform/check_aliasing/body_mutator.rs

```diff
@@ -0,0 +1,258 @@
+use super::{
+    BasicBlock, BasicBlockIdx, Body, Local, LocalDecl, Mutability, Operand, Place,
+    Span, Statement, Terminator, TerminatorKind, Ty, UnwindAction, VarDebugInfo,
+};
+
+/// BodyMutator combines the data of the function body
+/// with "ghost" basic block and local data, allowing the user
+/// to instrument the original body with instructions in the
+/// "ghost" section while iterating over the original data from the
+/// function body.
+pub struct BodyMutator {
+    blocks: Vec<BasicBlock>,
+    locals: Vec<LocalDecl>,
+    arg_count: usize,
+    var_debug_info: Vec<VarDebugInfo>,
+    spread_arg: Option<Local>,
+    span: Span,
+
+    ghost_locals: Vec<LocalDecl>,
+    ghost_blocks: Vec<BasicBlock>,
+    ghost_statements: Vec<Statement>,
+}
+
+impl BodyMutator {
```

Comment:
This has the same purpose and functionality as `MutableBody`. I think the instrumentation needs to integrate with the existing framework instead.

Reply:
Also, I am not too confident this handles control flow or multiple basic blocks correctly. We need more tests for that.